api/embedded.go

```diff
@@ -6,41 +6,25 @@ import (
 )
 
 type EmbeddedApi struct {
-	client client.IClient
-
-	//Accelerator embedded.AcceleratorApi
-	Pillar   embedded.PillarApi
-	Plasma   embedded.PlasmaApi
-	Sentinel embedded.SentinelApi
-	Spork    embedded.SporkApi
-	Stake    embedded.StakeApi
-	Swap     embedded.SwapApi
-	Token    embedded.TokenApi
-}
-
-func (e *EmbeddedApi) SetClient(client client.IClient) {
-	e.client = client
-
-	//e.Accelerator.SetClient(client)
-	e.Pillar.SetClient(client)
-	e.Plasma.SetClient(client)
-	e.Sentinel.SetClient(client)
-	e.Spork.SetClient(client)
-	e.Stake.SetClient(client)
-	e.Swap.SetClient(client)
-	e.Token.SetClient(client)
+	Accelerator embedded.AcceleratorApi
+	Pillar      embedded.PillarApi
+	Plasma      embedded.PlasmaApi
+	Sentinel    embedded.SentinelApi
+	Spork       embedded.SporkApi
+	Stake       embedded.StakeApi
+	Swap        embedded.SwapApi
+	Token       embedded.TokenApi
 }
 
-func NewEmbeddedApi() EmbeddedApi {
-	e := EmbeddedApi{
-		//Accelerator: embedded.AcceleratorApi{},
-		Pillar:   embedded.PillarApi{},
-		Plasma:   embedded.PlasmaApi{},
-		Sentinel: embedded.SentinelApi{},
-		Spork:    embedded.SporkApi{},
-		Stake:    embedded.StakeApi{},
-		Swap:     embedded.SwapApi{},
-		Token:    embedded.TokenApi{},
+func NewEmbeddedApi(client client.Client) EmbeddedApi {
```

Comment:
Can just use `c` instead of `client`, might help to avoid namespacing surprises.

Reply:
yes makes sense
not sure why i wrote it all out